impacket/tds.py

```diff
@@ -698,71 +698,32 @@ def kerberosLogin(self, database, username, password='', domain='', hashes=None,
-                ccache = CCache.loadFile(os.getenv('KRB5CCNAME'))
-            except:
-                # No cache present
-                pass
-            else:
-                # retrieve domain information from CCache file if needed
-                if domain == '':
-                    domain = ccache.principal.realm['data'].decode('utf-8')
-                    LOG.debug('Domain retrieved from CCache: %s' % domain)
-
-                LOG.debug("Using Kerberos Cache: %s" % os.getenv('KRB5CCNAME'))
-                principal = 'MSSQLSvc/%s.%s:%d@%s' % (self.server.split('.')[0], domain, self.port, domain.upper())
-                creds = ccache.getCredential(principal)
-
-                if creds is not None:
-                    TGS = creds.toTGS(principal)
-                    LOG.debug('Using TGS from cache')
-                else:
-                    # search for the port's instance name instead (instance name based SPN)
-                    LOG.debug('Searching target\'s instances to look for port number %s' % self.port)
-                    instances = self.getInstances()
-                    instanceName = None
-                    for i in instances:
-                        try:
-                            if int(i['tcp']) == self.port:
-                                instanceName = i['InstanceName']
-                        except:
-                            pass
-
-                    if instanceName:
-                        principal = 'MSSQLSvc/%s.%s:%s@%s' % (self.server, domain, instanceName, domain.upper())
-                        creds = ccache.getCredential(principal)
-
-                    if creds is not None:
-                        TGS = creds.toTGS(principal)
-                        LOG.debug('Using TGS from cache')
-                    else:
-                        # Let's try for the TGT and go from there
-                        principal = 'krbtgt/%s@%s' % (domain.upper(),domain.upper())
-                        creds =  ccache.getCredential(principal)
-                        if creds is not None:
-                            TGT = creds.toTGT()
-                            LOG.debug('Using TGT from cache')
-                        else:
-                            LOG.debug("No valid credentials found in cache. ")
-
-                # retrieve user information from CCache file if needed
-                if username == '' and creds is not None:
-                    username = creds['client'].prettyPrint().split(b'@')[0].decode('utf-8')
-                    LOG.debug('Username retrieved from CCache: %s' % username)
-                elif username == '' and len(ccache.principal.components) > 0:
-                    username = ccache.principal.components[0]['data'].decode('utf-8')
-                    LOG.debug('Username retrieved from CCache: %s' % username)
+        if useCache:
+            domain, username, _, TGS = CCache.parseFile(domain, username, 'MSSQLSvc/%s:%d' % (self.server, self.port))
```

Comment:
@0xdeaddood pointed out that here obtaining the TGT is required because it will be later on needed.